### PR TITLE
:gem:  :gemini:  Add basic UI :building_construction: 

### PR DESCRIPTION
**Pull request by SimonGideon**
## This PR includes:

- Build interactive console apps.

- Run a program using the command line. 

#### Thanks for reviewing my code :slightly_smiling_face:

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    @my_app.create_person('Teacher', age, name, specialization)
+  end
+
+  def create_student
+    puts 'Enter students age: '
+    age = gets.chomp.to_i
+    puts 'Enter students name: '
+    name = gets.chomp.to_s
+    @my_app.create_person('Student', age, name)
+  end
```
Comment (BenMukebo) [inline main.rb]: - [ ]  Hi @SimonGideon, great job your app is working. But you should make sure our school library console app behaves in the same way as in the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) video that was provided to us. 
- 1. When you create a person, the student (1) should be the first option and the teacher should be the second option (2) and not the inverse,
- 2. Make sure that in order to create the student  the app should request:
   - Age
   - Name
   - Has parent permission
- Kindly address this 💯 👍🏻  here is the screenshot of your console 👇🏻 

image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  def list_people
+    @people.each do |person|
+      case person
+      when Student
+        puts "[Student] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      when Teacher
+        puts "[Teacher] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
```
Comment (BenMukebo) [inline app.rb]: - [ ] There is a problem with the application when it comes to listing the people option (2) as you can see on the console 👇🏻 the name is displayed at the place of the age and the age is displayed where the Name age is. Please this 👍🏻

image

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    @my_app.gets_person
+    puts 'Enter book index:'
+    book_index = gets.chomp.to_i
+    puts 'Enter person Index:'
+    person_index = gets.chomp.to_i
+    puts 'Enter date(YYYY-MM-DD): '
+    date_input = gets.chomp
+    date = Date.parse(date_input)
+    @my_app.create_rental(book_index, person_index, date)
+  end
```
Comment (BenMukebo) [inline main.rb:116]: - [ ] The handle_create_rental method option (5) does not work as expected actually I'm getting this error message. 👇🏻  Please fix this 👍🏻 

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @SimonGideon,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (SimonGideon) [reply main.rb]: @BenMukebo  Noted the requested changes are implemented. Thanks for this :muscle: :slightly_smiling_face:

File: app.rb (hunk 2 above)
Comment (SimonGideon) [reply app.rb]: This fishy stuff:laughing: Lol! Now corrected and fine

File: main.rb (hunk 3 above)
Comment (SimonGideon) [reply main.rb:116]: Hey, @BenMukebo There is no person with `ID: 30`. Please choose a person that exits from the list provided by the app. It's nice to explore the drawbacks of my application. The above error will be handled on future features. :laughing: Thanks for the Review. :muscle: 

giphy (1)

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @SimonGideon , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: no linter errors 
:+1: app is working 
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SimonGideon** [comment]
@MeqdamAlqudah Thanks for your review :muscle: 
giphy

File: main.rb (hunk 3 above)
Comment (BenMukebo) [reply main.rb:116]: Yeah sure 😄

**Pull request merged**